### PR TITLE
SRCH-4469 Fix undefined method `search_engine`

### PR DESCRIPTION
**Pull request by stevenbarragan**
Do not raise errors when affiliate doesn't exists.

## Summary

Instead of accessing the `search_engine` property on an affiliate that can not be found this change leverages the database to check if an affiliate with the specified name and "SearchGov" as `search_engine` exist. If affiliate doesn't exist it returns false.

Issue: [here](https://cm-jira.usa.gov/browse/SRCH-4469).
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Review by lorawoodford — APPROVED** [review]
@stevenbarragan  Sorry I let this one linger so long - it evaded my filters since I don't think it was originally in the sprint, right?  

Anyway, this LGTM (once merge conflicts resolved).

That said, if this had come up in refinement/planning we could've discussed the fact that this Advanced Search functional is supposed to have been fully dead and deprecated in 2014 (per [confluence](https://cm.usa.gov/confluence/display/SRCH/The+Walking+Dead)).  If the frequency of errors is triggering your interest in this, I'd really love to see us scope out and prioritize actually removing what our documentation tells us is unused and dead code/functionality, rather than improving on/fixing bugs in code we say is unused and deprecated.

**Comment by stevenbarragan** [comment]
Thanks @lorawoodford. I had time and I just thought it was a quick fix but I should have bring it up first for planning. I'd love to remove code instead as well.

**Pull request merged**
